### ISSUE TITLE
Simplify MailAddressDisplayNameTest

### ISSUE THREAD

**Issue by ManickaP** [dotnet/runtime#39625]
Follow up of https://github.com/dotnet/runtime/pull/37874
Many of the tests in [MailAddressDisplayNameTest.cs](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressDisplayNameTest.cs) are duplicates of the same test, only with different inputs.
--> Change the tests from `[Fact]` to `[Theory]` and parametrize them with `[MemberData]`.

Some more clean up:
- some asserts have swapped expected and actual arguments: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressDisplayNameTest.cs#L44
- local variable with undesired underscore: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Mail/tests/Unit/MailAddressTests/MailAddressDisplayNameTest.cs#L63

Improvements to consider:
- reparse `MailAddress` from `ToString` result (depends on #37874)

cc: @jaymin93 @karelz @wfurt

**Comment by jaymin93** [comment]
should i do pr for that since i added test i am aware of test case and usage i think i can take this one and make it as expected

**Comment by ManickaP** [comment]
I'd first focus on finishing https://github.com/dotnet/runtime/pull/37874. After that, you can tackle this one if you wish.
